### ISSUE TITLE
How to update from Docker  v3.6.6 build 1 (gee2c67a) to v4?

### ISSUE THREAD

**Issue by ovizii** [snipe/snipe-it#4197]
I removed my old docker container (after backing everything vital up), pulled the latest snipe-it image, created a new container with exactly the same settings as the old container, started the container, visit my snipe-it dashboard and all I see is: `Whoops, looks like something went wrong. `

The error logs are similarly unhelpful: 
```
 docker logs snipeit
Module ssl disabled.
To activate the new configuration, you need to run:
  service apache2 restart
AH00558: apache2: Could not reliably determine the server's fully qualified domain name, using 172.17.0.4. Set the 'ServerName' directive globally to suppress this message
Module ssl already disabled
AH00558: apache2: Could not reliably determine the server's fully qualified domain name, using 172.17.0.4. Set the 'ServerName' directive globally to suppress this message
Module ssl already disabled
AH00558: apache2: Could not reliably determine the server's fully qualified domain name, using 172.17.0.4. Set the 'ServerName' directive globally to suppress this message

```

are there any instructions available for upgrading v3 to v4 via Docker image?

**Comment by furby** [comment]
https://snipe-it.readme.io/docs/upgrading-to-v4

I used the container ID to run the commands though I suspect you could (should?) just use the name as the ID changes.

`docker exec -i -t e35b33d1010 php artisan up`
for example

**Comment by ovizii** [comment]
I had read that link but wasn't sure if it was necessary for the Docker image. Did you successfully upgrade your Docker installation of snipe-it by following those instructions? If you can confirm I'll go ahead and give them a try.

Thanks for the reply :-)

**Comment by furby** [comment]
Yes, I successfully upgraded my docker by following those instructions 👍 

The composer command(s) didn't run for me so I kind of avoided that. With the snipe and mysql data files on a local volume, it's trivial to back up so I just kept going. 
Made a typo the first time and just ran it again and currently have (obvious) errors in the logs and it's working fine.

I'm using Docker for Mac, so I don't know if it's a little different in other environments but, I did steps
1,3,4, (I pulled version 4.0.9 before hand `docker pull snipe/snipe-it:v4.0.9`), 7,9,10,11,12

If it doesn't work I can alway write down step by step what I did

**Comment by ovizii** [comment]
every step except step 11 is clear to me. Can you share where to find this .env file? 

Here is how I run the docker image:
`docker run -d -p 9191:80 --name="snipeit" --restart=unless-stopped --env-file=/mnt/snipe-it/snipe-env -v /mnt/snipe-it:/var/lib/snipeit snipe/snipe-it`

that seems to generate a .env file inside the container in /var/www/html but the details specified inside `--env-file=/mnt/snipe-it/snipe-env` are not reflected in there but seem to be passed onto mz container via env variables.

Sorry, I am lost here, hoping someone can help me sort this out, meanwhile back to the working version :-)

**Comment by furby** [comment]
Are you missing the link to the mysql database there?
`--link snipe-mysql:mysql`

Also, in the docs it has the env file location after the mount. Perhaps that all it is

`docker run -d -p 9191:80 --name="snipeit" -v /mnt/snipe-it:/var/lib/snipeit --restart=unless-stopped  --env-file=/mnt/snipe-it/snipe-env snipe/snipe-it`

**Comment by ovizii** [comment]
Thanks furby you've been a great help so far. Let me clarify your questions: 

I know how to edit the .env file inside the container, it just does not reflect the arguments I am passing to my container with `--env-file=/mnt/snipe-it/snipe-env` - the .env inside the container looks like this: 

```
# --------------------------------------------
# REQUIRED: BASIC APP SETTINGS
# --------------------------------------------
#APP_ENV=develop
#APP_DEBUG=false
#APP_KEY=Change_this_key_or_snipe_will_get_ya
#APP_URL=http://127.0.0.1:32782
#APP_TIMEZONE=US/Pacific
#APP_LOCALE=en


# --------------------------------------------
# REQUIRED: DATABASE SETTINGS
# --------------------------------------------
DB_CONNECTION=mysql
DB_HOST=${MYSQL_PORT_3306_TCP_ADDR}
DB_DATABASE=${MYSQL_DATABASE}
DB_USERNAME=${MYSQL_USER}
DB_PASSWORD=${MYSQL_PASSWORD}
DB_PREFIX=null
DB_DUMP_PATH='/usr/bin'


# --------------------------------------------
# REQUIRED: OUTGOING MAIL SERVER SETTINGS
# --------------------------------------------
MAIL_DRIVER=smtp
MAIL_HOST=${MAIL_PORT_587_TCP_ADDR}
MAIL_PORT=${MAIL_PORT_587_TCP_PORT}
...
```

I decided to skip the `--link snipe-mysql:mysql` I access my mysql server differently, see my `--env-file=/mnt/snipe-it/snipe-env ` for details.

Here comes my snipe-env file: 
![image](https://user-images.githubusercontent.com/837902/31482861-481e83d0-af2a-11e7-8cdb-dbdb39c50f42.png)

And if I do a `docker inspect snipeit-container` I see in the env section that my details are indeed being used, just not in the .env file
![image](https://user-images.githubusercontent.com/837902/31482881-605bee56-af2a-11e7-9ae9-06c524ebbbb9.png)


**###edit###**
So sorry, I made a typo, I have a problem with step 10 not 11 :-( 
its all about this section: 

> open your .env file and add a new field called LEGACY_APP_KEY= and add your v3.x Snipe-IT APP_KEY value there.

I can see how this was confusing, my problem was editing a seemingly useless .env file. Does my thread now make more sense? I'm wondering what the point is of editing this file and how. there does not seem to be any text editor inside the container. How would I apply these edits to the snipe-env file so the changes end up inside the container?

**Comment by furby** [comment]
I'd written a longer post but, maybe just keep it simple. Copy the env file out to your host and copy it back?

**Comment by dejaeghered** [comment]
I just spun up a container following the documentation and I notice that the .env file in the container is not reflecting my data in the run --env-file.  The mysql pdo is not working because of the incorrect template data inside the .env file.  So its not just an upgrade issue but rather something missing with the latest v4 container not generating the .env file from the --env-file param.  How was the .env  generated in v3 versions using --env-file ?

**Comment by dejaeghered** [comment]
My issue was caused by a typo with the linking that caused env variables to be prefixed MYQL (type)

**Comment by snipe** [comment]
This is a duplicate of #4101, but this ticket has more info so keeping this one as primary.

We haven't heard back in a bit, so I'm going to close this ticket for now, but will re-open it if you're still having issues.

![hackerman-hacking](https://user-images.githubusercontent.com/197404/31936003-27a7f6d8-b865-11e7-8f5e-eb03062925f4.gif)

**Comment by ovizii** [comment]
@HinchK you're the real deal ,-) 
I finally got around to do this migration and after following your last post in this thread I got it working. The only thing I need to add is that I kept seeing the "Whoops, looks like something went wrong." until I emptied my browser cache / tried a new browser.

Finally I'm also using the latest SnipeIT version!!! 

Thanks to everyone else too who attempted to help me out here.
